src/unix/docker/dockerfiles/build/alpine/Dockerfile

```diff
@@ -38,4 +40,29 @@ RUN dotnet help
 
 # CLR Instrumentation Engine Build Prerequisites
 RUN apk update \
-    && apk add --no-cache libxml2-dev
\ No newline at end of file
+    && apk add --no-cache libxml2-dev
+
+# Install newer version of cmake. We need to run make commands inside the directory
+# that it is extracted to, so WORKDIR is the easiest context to use.
+WORKDIR /usr/share/cmakeinst
+
+# Copy manifest json to the target so that we can extract the download url and
+# version of cmake
+COPY "components/cmake/cgmanifest.json" /usr/share/cmakeinst
+COPY "scripts/readattr.sh" /usr/share/cmakeinst
+RUN chmod 775 ./readattr.sh \
+    && cmakeurl=$(./readattr.sh cgmanifest.json 'DownloadUrl') \
```

Comment:
I don't know how much more you want to iterate on this but a more concise and widely used option might be to use the [jq](https://pkgs.alpinelinux.org/package/v3.6/main/x86_64/jq) command-line tool.

Reply:
By concise, I mean you won't need a custom shell script to do the parsing.